NoSoft-SA/nspack
# Feature/cfg auto pltzr queue app

Summary of this change:
[See spec for reference](https://docs.google.com/document/d/1UrJGrIXAPRPgMLedp6aMokFGgwZZZ0ZkV625zAdhWTo/edit#heading=h.nuuh4cc83hr0)

Added three new pages for managing the allocations of an auto palletizer, located at Production | Auto Palletizing :
 - Auto Palletizer Queues
 - Manual Palletizing Queues
 - Active Products Dashboard

Together these three pages allow the user to allocate, activate, deactivate and purge a given- or all palletizer products for a specific palletizer queue, or all queues feeding from a specific line.

Users can be allocated to a given line via Development | Master files | Users | (via act) Edit -> `Packhouse Line.`
If a user is allocated to a `packhouse line`, then the should only be able to manage queues applicable to that line.
If left unspecified, the user is prompted with selecting a line before continuing to the manage screens.


CHANGELOG entry:

### Added
- Pages to applicable to Auto Palletizing under Production:
  - Auto Palletizer Queues
  - Manual Palletizing Queues
  - Active Products Dashboard.
- API calls for `purge queue(s)` and `get carton quantities on all queues` with mock API for responses.
### Changed
- `auto_palletizing/carton_verification` API call to additionally return the applicable `Queue`.
-  `auto_palletizing/carton_verification` API `target_market_group` changed to `target_market.`

How to test this code via the UI:
Palletizer Products are derived from the **current active production run** on load of either the manual/auto managing pages.
Ensure the Auto Palletizer Mock API is running to test Purge actions.
Ensure the Plant resources are present for the auto and manual palletizer queues.

Test the actual product to queue allocations with `curl -v --data '{"barcode":"784837"}' --header 'Content-Type: application/json' http://192.168.50.54/api/production/auto_palletizing/carton_verification`
but be sure to use an appropriate barcode value (carton label id)
